### ISSUE TITLE
FAILED: GzipSpec

### ISSUE THREAD

**Issue by ktoso** [akka/akka#20626]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/6317/consoleFull

```
[info] - should be able to decode chunk-by-chunk (depending on input chunks) *** FAILED *** (4 seconds, 558 milliseconds)
[info]   java.util.concurrent.TimeoutException: Futures timed out after [3 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:153)
[info]   at scala.concurrent.Await$$anonfun$ready$1.apply(package.scala:169)
[info]   at scala.concurrent.Await$$anonfun$ready$1.apply(package.scala:169)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]   at scala.concurrent.Await$.ready(package.scala:169)
[info]   at akka.http.impl.util.package$AddFutureAwaitResult.awaitResult(package.scala:60)
[info]   at akka.http.scaladsl.coding.CoderSpec$$anonfun$1$$anonfun$apply$mcV$sp$20.apply$mcV$sp(CoderSpec.scala:148)
[info]   at akka.http.scaladsl.coding.CoderSpec$$anonfun$1$$anonfun$apply$mcV$sp$20.apply(CoderSpec.scala:132)
[info]   at akka.http.scaladsl.coding.CoderSpec$$anonfun$1$$anonfun$apply$mcV$sp$20.apply(CoderSpec.scala:132)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at org.scalatest.WordSpec.withFixture(WordSpec.scala:1881)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:950)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:962)
[info]   at org.scalatest.WordSpec.runTest(WordSpec.scala:1881)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1021)
[info]   at org.scalatest.WordSpec.runTests(WordSpec.scala:1881)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1424)
[info]   at org.scalatest.WordSpec.org$scalatest$WordSpecLike$$super$run(WordSpec.scala:1881)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1067)
[info]   at akka.http.scaladsl.coding.CoderSpec.org$scalatest$BeforeAndAfterAll$$super$run(CoderSpec.scala:26)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:257)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:256)
[info]   at akka.http.scaladsl.coding.CoderSpec.run(CoderSpec.scala:26)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
[info]   at sbt.TestRunner.runTest$1(TestFramework.scala:76)
[info]   at sbt.TestRunner.run(TestFramework.scala:85)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
[info]   at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
[info]   at sbt.TestFunction.apply(TestFramework.scala:207)
[info]   at sbt.Tests$.sbt$Tests$$processRunnable$1(Tests.scala:239)
[info]   at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:245)
[info]   at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:245)
[info]   at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[info]   at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
[info]   at sbt.std.Transform$$anon$4.work(System.scala:63)
[info]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[info]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
[info]   at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[info]   at sbt.Execute.work(Execute.scala:237)
[info]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[info]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[info]   at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[info]   at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
[info] - should decode concatenated compressions (11 milliseconds)
[info] - should provide a better compression ratio than the standard Gzip/Gunzip streams (1 millisecond)
[info] - should throw an error on truncated input (4 milliseconds)
[info] - should throw an error if compressed data is just missing the trailer at the end (5 milliseconds)
[ERROR] [05/25/2016 13:14:56.539] [GzipSpec-akka.actor.default-dispatcher-5] [akka://GzipSpec/user/StreamSupervisor-38/flow-42-0-unknown-operation] Error in stage [akka.http.scaladsl.coding.GzipDecompressor@4a6cfda9]: Not in GZIP format
java.util.zip.ZipException: Not in GZIP format
	at akka.http.scaladsl.coding.GzipDecompressor$$anon$1.akka$http$scaladsl$coding$GzipDecompressor$$anon$$fail(Gzip.scala:94)
	at akka.http.scaladsl.coding.GzipDecompressor$$anon$1$ReadHeaders$.parse(Gzip.scala:79)
	at 
```

**Comment by drewhk** [comment]
I looked at this at detail, but I can't imagine so far how it can fail. I used some throttling+delay to poke on this (enable the fuzzer, too) with no avail so far. Recent execution times for this test case are around 200 ms, so a timeout of 3.seconds need an actual 10x slowdown, or a larger hiccup. While the test generates test chunks that can be larger or smaller depending on random choice, I am not sure it can cause this difference.

I keep this open for now.
